build/Common.props

```diff
@@ -1,30 +1,32 @@
     <MajorProductVersion Condition="$(MajorProductVersion) == ''">1</MajorProductVersion>
     <MinorProductVersion Condition="$(MinorProductVersion) == ''">0</MinorProductVersion>
     <PatchProductVersion Condition="$(PatchProductVersion) == ''">0</PatchProductVersion>
-    <VersionPrefix>$(MajorProductVersion).$(MinorProductVersion).$(PatchProductVersion)</VersionPrefix>
-    <VersionSuffix>preview5</VersionSuffix>
+    <VersionPrefix Condition="$(VersionPrefix) == ''">$(MajorProductVersion).$(MinorProductVersion).$(PatchProductVersion)</VersionPrefix>
+    <VersionSuffix Condition="$(VersionSuffix) == ''">preview5</VersionSuffix>
     <IsLocalBuild Condition=" $(IsLocalBuild) == ''">True</IsLocalBuild>
     <BuildNumber Condition=" $(IsLocalBuild) == 'True'">local$([System.DateTime]::Now.ToString(yyyyMMddHHmm))</BuildNumber>
-    <Version Condition=" '$(BuildNumber)' != '' ">$(VersionPrefix)-$(VersionSuffix)-$(BuildNumber)</Version>
+    <Version Condition=" '$(BuildNumber)' != '' And $(Version) == ''">$(VersionPrefix)-$(VersionSuffix)-$(BuildNumber)</Version>
     <Version Condition=" '$(Version)' == '' ">$(VersionPrefix)-$(VersionSuffix)</Version>
-    <AssemblyVersion>$(MajorProductVersion).$(MinorProductVersion).$(PatchProductVersion).0</AssemblyVersion>
-    <FileVersion>$(Version)</FileVersion>
+    <AssemblyVersion Condition="$(AssemblyVersion) == ''">$(VersionPrefix).0</AssemblyVersion>
+    <FileVersion  Condition="$(FileVersion) == ''">$(AssemblyVersion)</FileVersion>
     <CommitHash Condition="$(CommitHash) == ''">N/A</CommitHash>
-    <InformationalVersion>$(Version) Commit hash: $(CommitHash)</InformationalVersion>
-    <Authors>Microsoft</Authors>
-    <Company>Microsoft</Company>
-    <Copyright>© Microsoft Corporation. All rights reserved.</Copyright>
-    <RepositoryUrl>https://github.com/Azure/azure-functions-dotnet-worker</RepositoryUrl>
-    <RepositoryType>git</RepositoryType>
-    <PackageRequireLicenseAcceptance>True</PackageRequireLicenseAcceptance>
-    <PackageLicenseUrl>https://go.microsoft.com/fwlink/?linkid=2028464</PackageLicenseUrl>
-    <PackageProjectUrl>https://github.com/Azure/azure-functions-dotnet-worker</PackageProjectUrl>
-    <Nullable>enable</Nullable>
+    <InformationalVersion  Condition="$(InformationalVersion) == ''">$(Version) Commit hash: $(CommitHash)</InformationalVersion>
+    <LangVersion>preview</LangVersion>
+    <Nullable Condition="$(Nullable) == ''">enable</Nullable>
+    <SignAssembly>true</SignAssembly>
+    <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)/../key.snk</AssemblyOriginatorKeyFile>
     <DebugType>embedded</DebugType>
+    <EmbedUntrackedSources>true</EmbedUntrackedSources>
+    <ContinuousIntegrationBuild Condition="'$(TF_BUILD)' == 'true'">true"</ContinuousIntegrationBuild>
+    <Deterministic>true</Deterministic>
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All"/>
```

Comment:
Curious -- how does this know what repo we're in? Does it use `RepositoryUrl`?

Reply:
Primarily, yes. You can inspect the targets and tasks here to see the detailed flow: https://github.com/dotnet/sourcelink/tree/master/src/SourceLink.GitHub